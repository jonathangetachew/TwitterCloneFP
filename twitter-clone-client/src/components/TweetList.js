import React from 'react';
import {
    View,
    StyleSheet,
    FlatList,
    ActivityIndicator
} from 'react-native';
import { connect } from 'react-redux';
import { getTweets } from '../redux/actions/tweets';
import { Text } from "react-native-elements";
import TweetCard from './TweetCard';

export class TweetList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            dataSource: [],
            isLoading: true
        }
    }

    componentDidMount() {
        this.props.getTweets();
    }

    renderItem = ({ item }) => {
        return (
            <TweetCard {...item} />
        );
    }

    renderFixedHeader = () => {
        const { title } = this.props;
        return title ? (
            <View style={styles.header_style}>
                <Text h2 h2Style={{ marginTop: 10, marginBottom: 3, paddingBottom: 3, fontWeight: "700", fontSize: 22, paddingLeft: 5 }}>{title}</Text>
                <View style={{ width: '100%', height: 1, backgroundColor: '#ccc' }} />
            </View>
        ) : null;
    }
    renderSeperator = () => {
        return (
            <View style={{ width: '100%', height: 1, backgroundColor: '#ccc', marginVertical: 5 }} />
        );
    }

    handleLoadMore = () => {
        console.log("Warn load more");
    }
    render() {
        const { tweetData, loading, error } = this.props;
        if (loading) return <ActivityIndicator />
        return (
            <View style={styles.container}>
                <FlatList
                    data={tweetData}
                    renderItem={this.renderItem}
                    keyExtractor={(item, id) => id.toString()}
                    ItemSeparatorComponent={this.renderSeperator}
                    ListHeaderComponent={this.renderFixedHeader}
                    stickyHeaderIndices={[0]}
                    onEndReached={this.handleLoadMore}
                    onEndReachedThreshold={0}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        backgroundColor: '#FFF',
        maxWidth: 600,
        borderLeftWidth: 1,
        borderLeftColor: '#e6ecf0',
        borderRightWidth: 1,
        borderRightColor: '#e6ecf0',
    },
    profileimage: {
        width: 50,
        height: 50,
        borderRadius: 25,
        marginHorizontal: 5,
        marginTop: 5,
        overflow: "hidden",
    },
    infocontainer: {
        flex: 1,
        justifyContent: 'flex-start'
    },
    header: {
        marginBottom: 15,
        paddingBottom: 15
    },
    header_style: {
        backgroundColor: 'white',
        borderBottomWidth: 1,
        borderBottomColor: '#CCC',
        paddingHorizontal: 2
    },
    subtext: {
        marginRight: 10,
        color: '#657786',
        fontSize: 15,
        position: "absolute",
        top: -2,
        left: 20
    },
    iconView: {
        marginLeft: 1,
        flex: 1,
        flexDirection: "row"
    }
});

export default connect(({ tweets }) => ({
    tweetData: Object.values(tweets.data),
    loading: tweets.loading
}), dispatch => ({
    getTweets: () => dispatch(getTweets())
}))(TweetList);