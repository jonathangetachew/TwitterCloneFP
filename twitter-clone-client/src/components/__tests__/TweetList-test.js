import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import { TweetList } from '../TweetList';

const props = {
    getTweets: jest.fn()
}

jest.mock('../TweetCard', () => 'TweetCard');

describe('TweetList', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders the title`, () => {
        const tree = renderer.create(<TweetList title="Title" {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders without the title`, () => {
        const tree = renderer.create(<TweetList {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders render item`, () => {
        const tree = renderer.create(new TweetList().renderItem({ item: { name: 'Name' } })).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders render fixed header`, () => {
        const tree = renderer.create(new TweetList({ title: "Title" }).renderFixedHeader()).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders render separator`, () => {
        const tree = renderer.create(new TweetList().renderSeperator()).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders render separator`, () => {
        expect(new TweetList().handleLoadMore()).toEqual(undefined);
    });
});
