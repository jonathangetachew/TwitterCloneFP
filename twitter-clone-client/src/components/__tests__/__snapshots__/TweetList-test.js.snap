// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetList renders render fixed header 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderBottomColor": "#CCC",
      "borderBottomWidth": 1,
      "paddingHorizontal": 2,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "700",
        "marginBottom": 3,
        "marginTop": 10,
        "paddingBottom": 3,
        "paddingLeft": 5,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Title
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#ccc",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`TweetList renders render item 1`] = `
<TweetCard
  name="Name"
/>
`;

exports[`TweetList renders render separator 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ccc",
      "height": 1,
      "marginVertical": 5,
      "width": "100%",
    }
  }
/>
`;

exports[`TweetList renders the title 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderLeftColor": "#e6ecf0",
      "borderLeftWidth": 1,
      "borderRightColor": "#e6ecf0",
      "borderRightWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "maxWidth": 600,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "#CCC",
              "borderBottomWidth": 1,
              "paddingHorizontal": 2,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "fontWeight": "700",
                "marginBottom": 3,
                "marginTop": 10,
                "paddingBottom": 3,
                "paddingLeft": 5,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Title
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#ccc",
                "height": 1,
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`TweetList renders without the title 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderLeftColor": "#e6ecf0",
      "borderLeftWidth": 1,
      "borderRightColor": "#e6ecf0",
      "borderRightWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "maxWidth": 600,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</View>
`;
